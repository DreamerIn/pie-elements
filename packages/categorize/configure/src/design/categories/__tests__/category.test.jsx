import { shallow } from 'enzyme';
import React from 'react';
import { Category } from '../category';

describe('category', () => {
  let w;

  let onChange = jest.fn();
  let onDelete = jest.fn();
  let onDeleteChoice = jest.fn();
  let onAddChoice = jest.fn();

  const wrapper = extras => {
    const defaults = {
      classes: {},
      className: 'className',
      category: {
        id: '1',
        label: 'Category title'
      },
      onChange,
      onDelete,
      onDeleteChoice,
      onAddChoice
    };
    const props = { ...defaults, ...extras };
    return shallow(<Category {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('changeLabel', () => {
      it('calls onChange', () => {
        w = wrapper();
        w.instance().changeLabel('new label');
        expect(onChange).toBeCalledWith({ id: '1', label: 'new label' });
      });
    });
  });
});
