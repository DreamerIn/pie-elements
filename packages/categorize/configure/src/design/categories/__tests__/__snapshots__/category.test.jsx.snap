// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category snapshot renders 1`] = `
<WithStyles(Card)
  className="className"
>
  <WithStyles(InputHeader)
    label="Category title"
    onChange={[Function]}
    onDelete={[MockFunction]}
  />
  <Wrapped
    categoryId="1"
    onDeleteChoice={[MockFunction]}
    onDropChoice={[MockFunction]}
  />
  <WithStyles(CardActions)>
    <WithStyles(Component)
      label="delete"
      onClick={[MockFunction]}
    />
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
